shopping_mall/backend/ai/agent/executor.py

```
@@ -286,12 +329,28 @@ async def _run_loop(
+                (i, tc) for i, tc in enumerate(response.tool_calls)
+                if tc.name not in _RAG_TOOLS
+            ]
+
+            timed_results: list[tuple[ToolCall, str, int]] = [None] * len(response.tool_calls)  # type: ignore[list-item]
+
+            if rag_indexed:
+                rag_results = await asyncio.gather(*(
+                    self._timed_dispatch(tc, db, user_id, session_id)
+                    for _, tc in rag_indexed
+                ))
+                for (i, _), res in zip(rag_indexed, rag_results):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)